src/common/entity/compute_state_display.ts

```diff
@@ -29,37 +29,69 @@ export const computeStateDisplay = (
   const domain = computeStateDomain(stateObj);
 
   if (domain === "input_datetime") {
-    let date: Date;
-    if (!stateObj.attributes.has_time) {
+    if (state) {
+      // If trying to display an explicit state, need to parse the explict state to `Date` then format.
+      // Attributes aren't available, we have to use `state`.
+      try {
+        if (!stateObj.attributes.has_time) {
+          // Only has date.
+          const dateObj = new Date(state);
```

Comment:
I think we should make the `state` string a valid ISO 8601 string, by adding a date or time, then we don't have to struggle with all these issues, and don't have to rely on the current attributes.